core/src/cats/sql_create.cc

```diff
@@ -102,21 +102,25 @@ bool BareosDb::CreateJobRecord(JobControlRecord* jcr, JobDbRecord* jr)
  */
 bool BareosDb::CreateJobmediaRecord(JobControlRecord* jcr, JobMediaDbRecord* jm)
 {
-  int count;
-  char ed1[50], ed2[50], ed3[50];
-
   DbLocker _{this};
 
   /* clang-format off */
   Mmsg(cmd,
-       "UPDATE JobMedia SET LastIndex=%lu, EndFile=%lu, EndBlock=%lu, JobBytes=%s "
-       "WHERE JobId=%s AND MediaId=%s",
+       "UPDATE JobMedia SET "
+       "FirstIndex = (CASE WHEN firstindex=0 THEN %lu ELSE firstindex END), "
+       "StartBlock = (CASE WHEN startblock=0 THEN %lu ELSE startblock END), "
+       "StartFile  = (CASE WHEN startfile=0 THEN %lu ELSE startfile END), "
```

Comment:
I just had an idea...
We could make the zero-Record contain NULL instead of 0. That would still provide the purge-protection we want, but we could easily check if it is a zero-record here.
So we could then alter the WHERE of this UPDATE here to also contain `FirstIndex IS NOT NULL`, so it won't update the zero-record, eventually creating a new one.

Reply:
yeah... as discussed face-to-face: just forget it